lib/delayed_job_worker_pool/worker_pool.rb

```diff
@@ -114,14 +120,23 @@ def invoke_callback(callback_name, *args)
       options[callback_name].call(*args) if options[callback_name]
     end
 
-    def fork_worker
-      worker_pid = Kernel.fork { run_worker }
+    def fork_workers
+      options.fetch(:worker_groups).each do |group|
+        workers = group.workers || DEFAULT_WORKER_COUNT
+        worker_options = group.dj_worker_options
+        workers.times { fork_worker(worker_options) }
+      end
+    end
+
+    def fork_worker(options)
+      worker_pid = Kernel.fork { run_worker(options) }
       worker_pids << worker_pid
+      worker_pid_options[worker_pid] = options
       log("Started worker #{worker_pid}")
```

Comment:
Would it make sense to give worker groups names so we can include that information in log statements like this to provide visibility into which groups are churning workers?